Code/Editor/Controls/ConsoleSCB.cpp

```diff
@@ -205,6 +220,27 @@ bool ConsoleLineEdit::event(QEvent* ev)
 
 void ConsoleLineEdit::keyPressEvent(QKeyEvent* ev)
 {
+#if defined(CARBONATED)
```

Comment:
The way that if-endif preprocessor directives break the function bodies is not good. I prefer having complete functions inside preprocessor condition if possible. I do not see a perfect solution without massive code duplication, however there is a mitigation.
I suggest moving 
void ConsoleLineEdit::keyPressEvent(QKeyEvent* ev)
{
below
#if defined(CARBONATED)
Then add #else, duplicate 
void ConsoleLineEdit::keyPressEvent(QKeyEvent* ev)
{
This is a small code duplication with readability improvement benefit.